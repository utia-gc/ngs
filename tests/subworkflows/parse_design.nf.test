nextflow_workflow {

    name "Test Workflow Parse_Design"
    script "subworkflows/parse_design.nf"
    workflow "Parse_Design"

    test("Parse_Design successfully creates read channels from single-end samplesheet.") {

        when {
            params {
                publishDirReports = "${outputDir}/reports"
                publishDirData    = "${outputDir}/data"
            }
            workflow {
                """
                input[0] = "https://github.com/trev-f/SRAlign-test/raw/rnaseq/samplesheets/samplesheet_se.csv"
                """
            }
        }

        then {
            assert workflow.success
            assert workflow.out.samples.size() == 2
            assert snapshot(workflow.out).match()
        }

    }

    test("Parse_Design successfully creates read channels from paired-end samplesheet.") {

        when {
            params {
                publishDirReports = "${outputDir}/reports"
                publishDirData    = "${outputDir}/data"
            }
            workflow {
                """
                input[0] = "https://github.com/trev-f/SRAlign-test/raw/rnaseq/samplesheets/samplesheet_pe.csv"
                """
            }
        }

        then {
            assert workflow.success
            assert workflow.out.samples.size() == 2
            assert snapshot(workflow.out).match()
        }

    }

    test("Parse_Design successfully creates read channels from mixed single-end and paired-end samplesheet.") {

        when {
            params {
                publishDirReports = "${outputDir}/reports"
                publishDirData    = "${outputDir}/data"
            }
            workflow {
                """
                input[0] = "https://github.com/trev-f/SRAlign-test/raw/rnaseq/samplesheets/samplesheet_mix.csv"
                """
            }
        }

        then {
            assert workflow.success
            assert workflow.out.samples.size() == 4
            assert snapshot(workflow.out).match()
        }

    }

    test("Parse_Design successfully creates read channels with lanes and sample numbers -- SE + PE reads.") {

        when {
            params {
                publishDirReports = "${outputDir}/reports"
                publishDirData    = "${outputDir}/data"
            }
            workflow {
                """
                input[0] = 'https://github.com/utia-gc/ngs-test/raw/ngs/samplesheets/samplesheet_lanes_mix.csv'
                """
            }
        }

        then {
            // test workflow success
            assert workflow.success

            // test expected number samples
            assert workflow.out.samples.size() == 4

            // test reproducibility
            assert snapshot(workflow.out).match()
        }

    }

}
