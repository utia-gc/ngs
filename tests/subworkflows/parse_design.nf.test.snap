{
    "successfully creates channels from single-end samplesheet.": {
        "content": [
            {
                "0": [
                    [
                        {
                            "sampleName": "SRR1066657_GSM1299413_WT_NR_A",
                            "readType": "single"
                        },
                        [
                            "/trev-f/SRAlign-test/raw/rnaseq/data/reads/SRR1066657_GSM1299413_WT_NR_A_Saccharomyces_cerevisiae_RNA-Seq_50000.fastq.gz"
                        ]
                    ],
                    [
                        {
                            "sampleName": "SRR1066658_GSM1299414_WT_NR_B",
                            "readType": "single"
                        },
                        [
                            "/trev-f/SRAlign-test/raw/rnaseq/data/reads/SRR1066658_GSM1299414_WT_NR_B_Saccharomyces_cerevisiae_RNA-Seq_50000.fastq.gz"
                        ]
                    ]
                ],
                "samples": [
                    [
                        {
                            "sampleName": "SRR1066657_GSM1299413_WT_NR_A",
                            "readType": "single"
                        },
                        [
                            "/trev-f/SRAlign-test/raw/rnaseq/data/reads/SRR1066657_GSM1299413_WT_NR_A_Saccharomyces_cerevisiae_RNA-Seq_50000.fastq.gz"
                        ]
                    ],
                    [
                        {
                            "sampleName": "SRR1066658_GSM1299414_WT_NR_B",
                            "readType": "single"
                        },
                        [
                            "/trev-f/SRAlign-test/raw/rnaseq/data/reads/SRR1066658_GSM1299414_WT_NR_B_Saccharomyces_cerevisiae_RNA-Seq_50000.fastq.gz"
                        ]
                    ]
                ]
            }
        ],
        "timestamp": "2023-09-28T16:04:46+0000"
    },
    "Parse_Design successfully creates read channels with lanes and sample numbers -- SE + PE reads.": {
        "content": [
            {
                "0": [
                    [
                        {
                            "sampleName": "SRR6924569",
                            "readType": "paired",
                            "sampleNumber": "1",
                            "lane": "001"
                        },
                        [
                            "/utia-gc/ngs-test/raw/ngs/data/reads/raw/SRR6924569_S1_L001_R1_001.fastq.gz",
                            "/utia-gc/ngs-test/raw/ngs/data/reads/raw/SRR6924569_S1_L001_R2_001.fastq.gz"
                        ]
                    ],
                    [
                        {
                            "sampleName": "SRR6924569",
                            "readType": "paired",
                            "sampleNumber": "1",
                            "lane": "002"
                        },
                        [
                            "/utia-gc/ngs-test/raw/ngs/data/reads/raw/SRR6924569_S1_L002_R1_001.fastq.gz",
                            "/utia-gc/ngs-test/raw/ngs/data/reads/raw/SRR6924569_S1_L002_R2_001.fastq.gz"
                        ]
                    ],
                    [
                        {
                            "sampleName": "SRR1066657",
                            "readType": "single",
                            "sampleNumber": "3",
                            "lane": "001"
                        },
                        [
                            "/utia-gc/ngs-test/raw/ngs/data/reads/raw/SRR1066657_S3_L001_R1_001.fastq.gz"
                        ]
                    ],
                    [
                        {
                            "sampleName": "SRR1066657",
                            "readType": "single",
                            "sampleNumber": "3",
                            "lane": "002"
                        },
                        [
                            "/utia-gc/ngs-test/raw/ngs/data/reads/raw/SRR1066657_S3_L002_R1_001.fastq.gz"
                        ]
                    ]
                ],
                "samples": [
                    [
                        {
                            "sampleName": "SRR6924569",
                            "readType": "paired",
                            "sampleNumber": "1",
                            "lane": "001"
                        },
                        [
                            "/utia-gc/ngs-test/raw/ngs/data/reads/raw/SRR6924569_S1_L001_R1_001.fastq.gz",
                            "/utia-gc/ngs-test/raw/ngs/data/reads/raw/SRR6924569_S1_L001_R2_001.fastq.gz"
                        ]
                    ],
                    [
                        {
                            "sampleName": "SRR6924569",
                            "readType": "paired",
                            "sampleNumber": "1",
                            "lane": "002"
                        },
                        [
                            "/utia-gc/ngs-test/raw/ngs/data/reads/raw/SRR6924569_S1_L002_R1_001.fastq.gz",
                            "/utia-gc/ngs-test/raw/ngs/data/reads/raw/SRR6924569_S1_L002_R2_001.fastq.gz"
                        ]
                    ],
                    [
                        {
                            "sampleName": "SRR1066657",
                            "readType": "single",
                            "sampleNumber": "3",
                            "lane": "001"
                        },
                        [
                            "/utia-gc/ngs-test/raw/ngs/data/reads/raw/SRR1066657_S3_L001_R1_001.fastq.gz"
                        ]
                    ],
                    [
                        {
                            "sampleName": "SRR1066657",
                            "readType": "single",
                            "sampleNumber": "3",
                            "lane": "002"
                        },
                        [
                            "/utia-gc/ngs-test/raw/ngs/data/reads/raw/SRR1066657_S3_L002_R1_001.fastq.gz"
                        ]
                    ]
                ]
            }
        ],
        "timestamp": "2023-09-28T16:04:46+0000"
    },
    "successfully creates read channels from single-end samplesheet.": {
        "content": [
            {
                "0": [
                    [
                        {
                            "sampleName": "SRR1066657_GSM1299413_WT_NR_A",
                            "readType": "single"
                        },
                        [
                            "/trev-f/SRAlign-test/raw/rnaseq/data/reads/SRR1066657_GSM1299413_WT_NR_A_Saccharomyces_cerevisiae_RNA-Seq_50000.fastq.gz"
                        ]
                    ],
                    [
                        {
                            "sampleName": "SRR1066658_GSM1299414_WT_NR_B",
                            "readType": "single"
                        },
                        [
                            "/trev-f/SRAlign-test/raw/rnaseq/data/reads/SRR1066658_GSM1299414_WT_NR_B_Saccharomyces_cerevisiae_RNA-Seq_50000.fastq.gz"
                        ]
                    ]
                ],
                "samples": [
                    [
                        {
                            "sampleName": "SRR1066657_GSM1299413_WT_NR_A",
                            "readType": "single"
                        },
                        [
                            "/trev-f/SRAlign-test/raw/rnaseq/data/reads/SRR1066657_GSM1299413_WT_NR_A_Saccharomyces_cerevisiae_RNA-Seq_50000.fastq.gz"
                        ]
                    ],
                    [
                        {
                            "sampleName": "SRR1066658_GSM1299414_WT_NR_B",
                            "readType": "single"
                        },
                        [
                            "/trev-f/SRAlign-test/raw/rnaseq/data/reads/SRR1066658_GSM1299414_WT_NR_B_Saccharomyces_cerevisiae_RNA-Seq_50000.fastq.gz"
                        ]
                    ]
                ]
            }
        ],
        "timestamp": "2023-09-28T16:04:46+0000"
    },
    "Parse_Design successfully creates read channels from single-end samplesheet.": {
        "content": [
            {
                "0": [
                    [
                        {
                            "sampleName": "SRR1066657_GSM1299413_WT_NR_A",
                            "readType": "single"
                        },
                        [
                            "/trev-f/SRAlign-test/raw/rnaseq/data/reads/SRR1066657_GSM1299413_WT_NR_A_Saccharomyces_cerevisiae_RNA-Seq_50000.fastq.gz"
                        ]
                    ],
                    [
                        {
                            "sampleName": "SRR1066658_GSM1299414_WT_NR_B",
                            "readType": "single"
                        },
                        [
                            "/trev-f/SRAlign-test/raw/rnaseq/data/reads/SRR1066658_GSM1299414_WT_NR_B_Saccharomyces_cerevisiae_RNA-Seq_50000.fastq.gz"
                        ]
                    ]
                ],
                "samples": [
                    [
                        {
                            "sampleName": "SRR1066657_GSM1299413_WT_NR_A",
                            "readType": "single"
                        },
                        [
                            "/trev-f/SRAlign-test/raw/rnaseq/data/reads/SRR1066657_GSM1299413_WT_NR_A_Saccharomyces_cerevisiae_RNA-Seq_50000.fastq.gz"
                        ]
                    ],
                    [
                        {
                            "sampleName": "SRR1066658_GSM1299414_WT_NR_B",
                            "readType": "single"
                        },
                        [
                            "/trev-f/SRAlign-test/raw/rnaseq/data/reads/SRR1066658_GSM1299414_WT_NR_B_Saccharomyces_cerevisiae_RNA-Seq_50000.fastq.gz"
                        ]
                    ]
                ]
            }
        ],
        "timestamp": "2023-09-28T16:04:46+0000"
    },
    "Parse_Design successfully creates read channels from mixed single-end and paired-end samplesheet.": {
        "content": [
            {
                "0": [
                    [
                        {
                            "sampleName": "SRR6924569_GSM3073206_Saccharomyces_cerevisiae-AR_Biological_Repeat-2",
                            "readType": "paired"
                        },
                        [
                            "/trev-f/SRAlign-test/raw/rnaseq/data/reads/SRR6924569_GSM3073206_Saccharomyces_cerevisiae-AR_Biological_Repeat-2_Saccharomyces_cerevisiae_RNA-Seq_1_50000.fastq.gz",
                            "/trev-f/SRAlign-test/raw/rnaseq/data/reads/SRR6924569_GSM3073206_Saccharomyces_cerevisiae-AR_Biological_Repeat-2_Saccharomyces_cerevisiae_RNA-Seq_2_50000.fastq.gz"
                        ]
                    ],
                    [
                        {
                            "sampleName": "SRR6924589_GSM3073211_Saccharomyces_cerevisiae-AN_Biological_Repeat-1",
                            "readType": "paired"
                        },
                        [
                            "/trev-f/SRAlign-test/raw/rnaseq/data/reads/SRR6924589_GSM3073211_Saccharomyces_cerevisiae-AN_Biological_Repeat-1_Saccharomyces_cerevisiae_RNA-Seq_1_50000.fastq.gz",
                            "/trev-f/SRAlign-test/raw/rnaseq/data/reads/SRR6924589_GSM3073211_Saccharomyces_cerevisiae-AN_Biological_Repeat-1_Saccharomyces_cerevisiae_RNA-Seq_2_50000.fastq.gz"
                        ]
                    ],
                    [
                        {
                            "sampleName": "SRR1066657_GSM1299413_WT_NR_A",
                            "readType": "single"
                        },
                        [
                            "/trev-f/SRAlign-test/raw/rnaseq/data/reads/SRR1066657_GSM1299413_WT_NR_A_Saccharomyces_cerevisiae_RNA-Seq_50000.fastq.gz"
                        ]
                    ],
                    [
                        {
                            "sampleName": "SRR1066658_GSM1299414_WT_NR_B",
                            "readType": "single"
                        },
                        [
                            "/trev-f/SRAlign-test/raw/rnaseq/data/reads/SRR1066658_GSM1299414_WT_NR_B_Saccharomyces_cerevisiae_RNA-Seq_50000.fastq.gz"
                        ]
                    ]
                ],
                "samples": [
                    [
                        {
                            "sampleName": "SRR6924569_GSM3073206_Saccharomyces_cerevisiae-AR_Biological_Repeat-2",
                            "readType": "paired"
                        },
                        [
                            "/trev-f/SRAlign-test/raw/rnaseq/data/reads/SRR6924569_GSM3073206_Saccharomyces_cerevisiae-AR_Biological_Repeat-2_Saccharomyces_cerevisiae_RNA-Seq_1_50000.fastq.gz",
                            "/trev-f/SRAlign-test/raw/rnaseq/data/reads/SRR6924569_GSM3073206_Saccharomyces_cerevisiae-AR_Biological_Repeat-2_Saccharomyces_cerevisiae_RNA-Seq_2_50000.fastq.gz"
                        ]
                    ],
                    [
                        {
                            "sampleName": "SRR6924589_GSM3073211_Saccharomyces_cerevisiae-AN_Biological_Repeat-1",
                            "readType": "paired"
                        },
                        [
                            "/trev-f/SRAlign-test/raw/rnaseq/data/reads/SRR6924589_GSM3073211_Saccharomyces_cerevisiae-AN_Biological_Repeat-1_Saccharomyces_cerevisiae_RNA-Seq_1_50000.fastq.gz",
                            "/trev-f/SRAlign-test/raw/rnaseq/data/reads/SRR6924589_GSM3073211_Saccharomyces_cerevisiae-AN_Biological_Repeat-1_Saccharomyces_cerevisiae_RNA-Seq_2_50000.fastq.gz"
                        ]
                    ],
                    [
                        {
                            "sampleName": "SRR1066657_GSM1299413_WT_NR_A",
                            "readType": "single"
                        },
                        [
                            "/trev-f/SRAlign-test/raw/rnaseq/data/reads/SRR1066657_GSM1299413_WT_NR_A_Saccharomyces_cerevisiae_RNA-Seq_50000.fastq.gz"
                        ]
                    ],
                    [
                        {
                            "sampleName": "SRR1066658_GSM1299414_WT_NR_B",
                            "readType": "single"
                        },
                        [
                            "/trev-f/SRAlign-test/raw/rnaseq/data/reads/SRR1066658_GSM1299414_WT_NR_B_Saccharomyces_cerevisiae_RNA-Seq_50000.fastq.gz"
                        ]
                    ]
                ]
            }
        ],
        "timestamp": "2023-09-28T16:04:46+0000"
    },
    "successfully creates read channels from paired-end samplesheet.": {
        "content": [
            {
                "0": [
                    [
                        {
                            "sampleName": "SRR6924569_GSM3073206_Saccharomyces_cerevisiae-AR_Biological_Repeat-2",
                            "readType": "paired"
                        },
                        [
                            "/trev-f/SRAlign-test/raw/rnaseq/data/reads/SRR6924569_GSM3073206_Saccharomyces_cerevisiae-AR_Biological_Repeat-2_Saccharomyces_cerevisiae_RNA-Seq_1_50000.fastq.gz",
                            "/trev-f/SRAlign-test/raw/rnaseq/data/reads/SRR6924569_GSM3073206_Saccharomyces_cerevisiae-AR_Biological_Repeat-2_Saccharomyces_cerevisiae_RNA-Seq_2_50000.fastq.gz"
                        ]
                    ],
                    [
                        {
                            "sampleName": "SRR6924589_GSM3073211_Saccharomyces_cerevisiae-AN_Biological_Repeat-1",
                            "readType": "paired"
                        },
                        [
                            "/trev-f/SRAlign-test/raw/rnaseq/data/reads/SRR6924589_GSM3073211_Saccharomyces_cerevisiae-AN_Biological_Repeat-1_Saccharomyces_cerevisiae_RNA-Seq_1_50000.fastq.gz",
                            "/trev-f/SRAlign-test/raw/rnaseq/data/reads/SRR6924589_GSM3073211_Saccharomyces_cerevisiae-AN_Biological_Repeat-1_Saccharomyces_cerevisiae_RNA-Seq_2_50000.fastq.gz"
                        ]
                    ]
                ],
                "samples": [
                    [
                        {
                            "sampleName": "SRR6924569_GSM3073206_Saccharomyces_cerevisiae-AR_Biological_Repeat-2",
                            "readType": "paired"
                        },
                        [
                            "/trev-f/SRAlign-test/raw/rnaseq/data/reads/SRR6924569_GSM3073206_Saccharomyces_cerevisiae-AR_Biological_Repeat-2_Saccharomyces_cerevisiae_RNA-Seq_1_50000.fastq.gz",
                            "/trev-f/SRAlign-test/raw/rnaseq/data/reads/SRR6924569_GSM3073206_Saccharomyces_cerevisiae-AR_Biological_Repeat-2_Saccharomyces_cerevisiae_RNA-Seq_2_50000.fastq.gz"
                        ]
                    ],
                    [
                        {
                            "sampleName": "SRR6924589_GSM3073211_Saccharomyces_cerevisiae-AN_Biological_Repeat-1",
                            "readType": "paired"
                        },
                        [
                            "/trev-f/SRAlign-test/raw/rnaseq/data/reads/SRR6924589_GSM3073211_Saccharomyces_cerevisiae-AN_Biological_Repeat-1_Saccharomyces_cerevisiae_RNA-Seq_1_50000.fastq.gz",
                            "/trev-f/SRAlign-test/raw/rnaseq/data/reads/SRR6924589_GSM3073211_Saccharomyces_cerevisiae-AN_Biological_Repeat-1_Saccharomyces_cerevisiae_RNA-Seq_2_50000.fastq.gz"
                        ]
                    ]
                ]
            }
        ],
        "timestamp": "2023-09-28T16:04:46+0000"
    },
    "successfully creates read channels from mixed single-end and paired-end samplesheet.": {
        "content": [
            {
                "0": [
                    [
                        {
                            "sampleName": "SRR6924569_GSM3073206_Saccharomyces_cerevisiae-AR_Biological_Repeat-2",
                            "readType": "paired"
                        },
                        [
                            "/trev-f/SRAlign-test/raw/rnaseq/data/reads/SRR6924569_GSM3073206_Saccharomyces_cerevisiae-AR_Biological_Repeat-2_Saccharomyces_cerevisiae_RNA-Seq_1_50000.fastq.gz",
                            "/trev-f/SRAlign-test/raw/rnaseq/data/reads/SRR6924569_GSM3073206_Saccharomyces_cerevisiae-AR_Biological_Repeat-2_Saccharomyces_cerevisiae_RNA-Seq_2_50000.fastq.gz"
                        ]
                    ],
                    [
                        {
                            "sampleName": "SRR6924589_GSM3073211_Saccharomyces_cerevisiae-AN_Biological_Repeat-1",
                            "readType": "paired"
                        },
                        [
                            "/trev-f/SRAlign-test/raw/rnaseq/data/reads/SRR6924589_GSM3073211_Saccharomyces_cerevisiae-AN_Biological_Repeat-1_Saccharomyces_cerevisiae_RNA-Seq_1_50000.fastq.gz",
                            "/trev-f/SRAlign-test/raw/rnaseq/data/reads/SRR6924589_GSM3073211_Saccharomyces_cerevisiae-AN_Biological_Repeat-1_Saccharomyces_cerevisiae_RNA-Seq_2_50000.fastq.gz"
                        ]
                    ],
                    [
                        {
                            "sampleName": "SRR1066657_GSM1299413_WT_NR_A",
                            "readType": "single"
                        },
                        [
                            "/trev-f/SRAlign-test/raw/rnaseq/data/reads/SRR1066657_GSM1299413_WT_NR_A_Saccharomyces_cerevisiae_RNA-Seq_50000.fastq.gz"
                        ]
                    ],
                    [
                        {
                            "sampleName": "SRR1066658_GSM1299414_WT_NR_B",
                            "readType": "single"
                        },
                        [
                            "/trev-f/SRAlign-test/raw/rnaseq/data/reads/SRR1066658_GSM1299414_WT_NR_B_Saccharomyces_cerevisiae_RNA-Seq_50000.fastq.gz"
                        ]
                    ]
                ],
                "samples": [
                    [
                        {
                            "sampleName": "SRR6924569_GSM3073206_Saccharomyces_cerevisiae-AR_Biological_Repeat-2",
                            "readType": "paired"
                        },
                        [
                            "/trev-f/SRAlign-test/raw/rnaseq/data/reads/SRR6924569_GSM3073206_Saccharomyces_cerevisiae-AR_Biological_Repeat-2_Saccharomyces_cerevisiae_RNA-Seq_1_50000.fastq.gz",
                            "/trev-f/SRAlign-test/raw/rnaseq/data/reads/SRR6924569_GSM3073206_Saccharomyces_cerevisiae-AR_Biological_Repeat-2_Saccharomyces_cerevisiae_RNA-Seq_2_50000.fastq.gz"
                        ]
                    ],
                    [
                        {
                            "sampleName": "SRR6924589_GSM3073211_Saccharomyces_cerevisiae-AN_Biological_Repeat-1",
                            "readType": "paired"
                        },
                        [
                            "/trev-f/SRAlign-test/raw/rnaseq/data/reads/SRR6924589_GSM3073211_Saccharomyces_cerevisiae-AN_Biological_Repeat-1_Saccharomyces_cerevisiae_RNA-Seq_1_50000.fastq.gz",
                            "/trev-f/SRAlign-test/raw/rnaseq/data/reads/SRR6924589_GSM3073211_Saccharomyces_cerevisiae-AN_Biological_Repeat-1_Saccharomyces_cerevisiae_RNA-Seq_2_50000.fastq.gz"
                        ]
                    ],
                    [
                        {
                            "sampleName": "SRR1066657_GSM1299413_WT_NR_A",
                            "readType": "single"
                        },
                        [
                            "/trev-f/SRAlign-test/raw/rnaseq/data/reads/SRR1066657_GSM1299413_WT_NR_A_Saccharomyces_cerevisiae_RNA-Seq_50000.fastq.gz"
                        ]
                    ],
                    [
                        {
                            "sampleName": "SRR1066658_GSM1299414_WT_NR_B",
                            "readType": "single"
                        },
                        [
                            "/trev-f/SRAlign-test/raw/rnaseq/data/reads/SRR1066658_GSM1299414_WT_NR_B_Saccharomyces_cerevisiae_RNA-Seq_50000.fastq.gz"
                        ]
                    ]
                ]
            }
        ],
        "timestamp": "2023-09-28T16:04:46+0000"
    },
    "Parse_Design successfully creates read channels from paired-end samplesheet.": {
        "content": [
            {
                "0": [
                    [
                        {
                            "sampleName": "SRR6924569_GSM3073206_Saccharomyces_cerevisiae-AR_Biological_Repeat-2",
                            "readType": "paired"
                        },
                        [
                            "/trev-f/SRAlign-test/raw/rnaseq/data/reads/SRR6924569_GSM3073206_Saccharomyces_cerevisiae-AR_Biological_Repeat-2_Saccharomyces_cerevisiae_RNA-Seq_1_50000.fastq.gz",
                            "/trev-f/SRAlign-test/raw/rnaseq/data/reads/SRR6924569_GSM3073206_Saccharomyces_cerevisiae-AR_Biological_Repeat-2_Saccharomyces_cerevisiae_RNA-Seq_2_50000.fastq.gz"
                        ]
                    ],
                    [
                        {
                            "sampleName": "SRR6924589_GSM3073211_Saccharomyces_cerevisiae-AN_Biological_Repeat-1",
                            "readType": "paired"
                        },
                        [
                            "/trev-f/SRAlign-test/raw/rnaseq/data/reads/SRR6924589_GSM3073211_Saccharomyces_cerevisiae-AN_Biological_Repeat-1_Saccharomyces_cerevisiae_RNA-Seq_1_50000.fastq.gz",
                            "/trev-f/SRAlign-test/raw/rnaseq/data/reads/SRR6924589_GSM3073211_Saccharomyces_cerevisiae-AN_Biological_Repeat-1_Saccharomyces_cerevisiae_RNA-Seq_2_50000.fastq.gz"
                        ]
                    ]
                ],
                "samples": [
                    [
                        {
                            "sampleName": "SRR6924569_GSM3073206_Saccharomyces_cerevisiae-AR_Biological_Repeat-2",
                            "readType": "paired"
                        },
                        [
                            "/trev-f/SRAlign-test/raw/rnaseq/data/reads/SRR6924569_GSM3073206_Saccharomyces_cerevisiae-AR_Biological_Repeat-2_Saccharomyces_cerevisiae_RNA-Seq_1_50000.fastq.gz",
                            "/trev-f/SRAlign-test/raw/rnaseq/data/reads/SRR6924569_GSM3073206_Saccharomyces_cerevisiae-AR_Biological_Repeat-2_Saccharomyces_cerevisiae_RNA-Seq_2_50000.fastq.gz"
                        ]
                    ],
                    [
                        {
                            "sampleName": "SRR6924589_GSM3073211_Saccharomyces_cerevisiae-AN_Biological_Repeat-1",
                            "readType": "paired"
                        },
                        [
                            "/trev-f/SRAlign-test/raw/rnaseq/data/reads/SRR6924589_GSM3073211_Saccharomyces_cerevisiae-AN_Biological_Repeat-1_Saccharomyces_cerevisiae_RNA-Seq_1_50000.fastq.gz",
                            "/trev-f/SRAlign-test/raw/rnaseq/data/reads/SRR6924589_GSM3073211_Saccharomyces_cerevisiae-AN_Biological_Repeat-1_Saccharomyces_cerevisiae_RNA-Seq_2_50000.fastq.gz"
                        ]
                    ]
                ]
            }
        ],
        "timestamp": "2023-09-28T16:04:46+0000"
    }
}